app/scripts/lib/util.ts

```diff
@@ -433,3 +435,38 @@ export const getMethodDataName = async (
 export function getBooleanFlag(value: string | boolean | undefined): boolean {
   return value === true || value === 'true';
 }
+
+type AssetsRatesState = {
+  metamask: MultichainAssetsRatesControllerState;
+};
+
+export function getConversionRatesForNativeAsset({
+  conversionRates,
+  chainId,
+}: {
+  conversionRates: AssetsRatesState['metamask']['conversionRates'];
+  chainId: string;
+}): { rate: number } | null {
+  // Return early if conversionRates is falsy
+  if (!conversionRates) {
+    return null;
+  }
+
+  let conversionRateResult = null;
+
+  Object.entries(conversionRates).forEach(
+    ([caip19Identifier, conversionRate]) => {
+      const parsedCaip19Identifier = parseCaipAssetType(
+        caip19Identifier as CaipAssetType,
+      );
+      if (
+        parsedCaip19Identifier.assetNamespace === 'slip44' &&
+        parsedCaip19Identifier.chainId === chainId
+      ) {
+        conversionRateResult = conversionRate;
+      }
+    },
+  );
+
+  return conversionRateResult;
+}
```

Comment:
I don't understand why we need this ?

Reply:
The `conversionRates` state contains both native tokens (such as SOL) and non-native tokens, organized by their `caip-19` identifiers. In contrast, the coin-overview displays only the balance of the native token. This function identifies the native token from the conversionRates based on its `caip-19` identifier and returns it for the selected network. Let me know if you need further clarification.

Reply:
The getMultichainConversionRate used to use the getMultichainCoinRates selector which was getting data from the legacy ratesController.
This one is getting the rates from the new assetsRatesController.

Note: this should not break BTC since we agreed to disable it here https://github.com/MetaMask/metamask-extension/pull/30477?email_source=slack

Reply:
Okay, now I understand. Sorry, the selector name consuming this is misleading as it's only getting the conversion rate for the native asset but nothing says this in the name. Anyways, this is not really related to the PR so it should be okay for now.